pipeline/mutate/mutator_id_token.go

```diff
@@ -108,16 +104,21 @@ func (a *MutatorIDToken) tokenFromCache(config *CredentialsIDTokenConfig, sessio
 }
 
 func (a *MutatorIDToken) tokenToCache(config *CredentialsIDTokenConfig, session *authn.AuthenticationSession, claims []byte, ttl time.Duration, expiresAt time.Time, token string) {
-	if !a.tokenCacheEnabled {
+	if !config.Cache.Enabled {
 		return
 	}
 
 	key := a.cacheKey(config, ttl, claims, session)
-	a.tokenCache.Set(key, &idTokenCacheContainer{
-		TTL:       ttl,
-		ExpiresAt: expiresAt,
-		Token:     token,
-	}, 0)
+	a.tokenCache.SetWithTTL(
+		key,
+		&idTokenCacheContainer{
+			TTL:       ttl,
+			ExpiresAt: expiresAt,
+			Token:     token,
+		},
+		0,
+		ttl,
+	)
```

Comment:
Another probably reason for the memory leak:

Keys are only deleted from cache when requested and expired.

I don't know if ristretto flushes other keys when the max cost is reached, but this could be another lead.

Reply:
We deployed this in production, and it fixed the memory leak indeed :)